src/gh-api-client.ts

```diff
@@ -12,20 +13,39 @@ import { validateObject } from "./json-validation";
 
 export const userAgent = "GitHub multi-repository variant analysis action";
 
-function getOctokitRequestInterface(): RequestInterface {
-  const octokit = new Octokit({ userAgent, retry });
-
-  const signedAuthToken = getSignedAuthToken();
-  return octokit.request.defaults({
-    request: {
-      hook: (request: RequestInterface, options: EndpointOptions) => {
-        if (options.headers) {
-          options.headers.authorization = `RemoteAuth ${signedAuthToken}`;
-        }
-        return request(options);
+function getOctokit(): Octokit {
```

Comment:
Thanks for cleaning up this interface! ❤️ 

The naming and returning that `RequestInterface` type was a bit awkward, but at the time I couldn't find a better way of doing things. What you've got here looks much better!

Reply:
Thank you! I spent a bit figuring out how to simulate adding the hook without returning `octokit.request.defaults`, which does not carry the rest of the features, but reversing a builtin [auth strategy](https://github.com/octokit/auth-token.js/blob/main/src/index.ts#L27) I saw it was almost the same implementation 🪄